src/Listeners/GenerateHelpers.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Barryvdh\LaravelIdeHelper\Listeners;
+
+use Illuminate\Console\Events\CommandFinished;
+use Illuminate\Contracts\Config\Repository as Config;
+use Illuminate\Contracts\Console\Kernel as Artisan;
+
+class GenerateHelpers
+{
+    /** @var \Illuminate\Contracts\Console\Kernel */
+    protected $artisan;
+
+    /** @var \Illuminate\Config\Repository */
+    protected $config;
+
+    /**
+     * @param \Illuminate\Contracts\Console\Kernel $artisan
+     * @param \Illuminate\Config\Repository $config
+     */
+    public function __construct(Artisan $artisan, Config $config)
+    {
+        $this->artisan = $artisan;
+        $this->config = $config;
+    }
+
+    /**
+     * Handle the event.
+     *
+     * @param  CommandFinished  $event
+     */
+    public function handle(CommandFinished $event)
+    {
+        switch ($event->command) {
+            case "package:discover":
+                $this->generate('generate', $event->output);
+                $this->generate('meta', $event->output);
+                break;
+            case "migrate":
+            case "migrate:rollback":
+                $this->generate('models', $event->output);
+                break;
+        }
+    }
+
```

Comment:
```suggestion

    /**
     * @param \Illuminate\Console\Events\CommandFinished $event
     *
     * @return bool
     */
    protected function eventIsInvalid(CommandFinished $event): bool
    {
        /*
         * The event is invalid if any of the following are true:
         * 1. exitCode is not zero (0)
         * 2. input does not have a "command" argument; this happens when the
         *    -V/--version flag is used
         * 3. input has "help" option (-h/--help)
         * 4. input has "pretend" option (--pretend)
         */
        return $event->exitCode !== 0 ||
            !$event->input->hasArgument("command") ||
            $this->hasOption($event, "help") ||
            $this->hasOption($event, "pretend");
    }

```

Reply:
You have a point, this should be mostly solved by using MigrationsEnded event as a trigger as mentioned above, nut it can be definitely used on the package discover command, because as far as I know there is no PackageDiscovered event. Thanks for the suggestions.